lib/json-schema/schema.rb

```diff
@@ -6,6 +6,8 @@ class Schema
     attr_accessor :schema, :uri, :validator
 
     def initialize(schema,uri,parent_validator=nil)
+      raise ArgumentError, "schema must be a hash; got: #{schema.inspect}" unless schema.is_a?(Hash)
```

Comment:
I'm trying (slowly) to move the codebase away from explicit type checking towards duck typing. Could you please alter this check from `schema.is_a?(Hash)` to `schema.respond_to?(:to_hash)`, and the line below from `@schema = schema` to `@schema = schema.to_hash`?